hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -965,6 +1015,24 @@ def ir_value_permitted(value, ir_type, kwargs):
     raise NotImplementedError(f"unhandled type {type(value)} of ir value {value}")
 
 
+def ir_value_equal(ir_type, v1, v2):
+    if ir_type != "float":
+        return v1 == v2
+    return float_to_int(v1) == float_to_int(v2)
+
+
+def ir_kwargs_equal(ir_type, kwargs1, kwargs2):
+    if ir_type != "float":
+        return kwargs1 == kwargs2
+    return (
+        float_to_int(kwargs1["min_value"]) == float_to_int(kwargs2["min_value"])
+        and float_to_int(kwargs1["max_value"]) == float_to_int(kwargs2["max_value"])
+        and kwargs1["allow_nan"] == kwargs2["allow_nan"]
+        and kwargs1["smallest_nonzero_magnitude"]
+        == kwargs2["smallest_nonzero_magnitude"]
+    )
+
```

Comment:
bit the bullet here. I have had to consciously stop myself from using equality for ir things in some cases lest it be subtly wrong for floats, and I just don't want to have to think about it anymore.

Reply:
`ir_value_equal` is definitely useful; `ir_kwargs_equal` is unnecessary because the bounds or smallest-magnitude can never be `nan`.

Reply:
ah, unfortunately `min_value=-0.0` vs `min_value=0.0` disagrees 😄

Reply:
Gah, of course.